internal/pkg/plugin/devlakeconfig/devlakeconfig.go

```diff
@@ -99,7 +178,66 @@ func DeleteConfig(options plugininstaller.RawOptions) error {
 }
 
 func UpdateConfig(options plugininstaller.RawOptions) error {
-	// TODO(daniel-hutao): implement later
+	opts, err := NewOptions(options)
+	if err != nil {
+		return err
+	}
+	cfg, err := configmanager.NewManager("config-devlake-config.yaml").LoadConfig()
+	if err != nil {
+		return err
+	}
+	smgr, _ := statemanager.NewManager(*cfg.State)
+	stateLakeConfig := smgr.GetState("devlake-config_default")
```

Comment:
"key" isn't the "devlake-config_default" here. Users can customise "instanceID" by themselves. Maybe the plugins shouldn't try to get the "State" and compare it to "Config". One way to implement "Update" is that the plugins get the "Resource state" and compare it to "Config" and then decide how to "Update".